### DISCUSSION TITLE
Python wrapper

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#472] [Dev notes]
We're seeking Python contributors to lead an effort to build a version-synced parallel path Python wrapper for this library.

Please refer to the separate Python repo for our initial release task list:
https://github.com/DaveSkender/Stock.Indicators.Python/issues/62

**Comment by DaveSkender** [comment]
@LeeDongGeon1996 if you or anyone is interested in [contributing](https://daveskender.github.io/Stock.Indicators/docs/CONTRIBUTING.html) to this, let us know and we can start to discuss implementation details.  I haven't done a lot of research, so I'm unclear on whether we need to add any more C# code to make this easier.  I'm definitely here to help and contribute in those areas if needed.  I was thinking we put all the independent Python wrapper code in a new `/wraps/python` folder in the repo.

My only ask is that we implement in small increments.  In other words, the first Pull Request should only include one or two indicators, without any documentation updates, and will be used to establish the baseline approach.

Some initial thoughts:

- Use [Python.NET](https://pypi.org/project/pythonnet) or a comparable integration approach for the wrapper
- Versioned with the primary NuGet package for consistency
- Release simultaneously to both NuGet and Python community package managers
- Use the common/shared documentation site, for consistency

We can then figure out a strategy to get all the way to a published package that people can get from the public Python package managers.  I'll start a Pull Request To-Do list above to capture the sequence of increments and who's working on what parts to coordinate amongst collaborators.

**To contribute, announce here what piece you're picking up and I'll put your name next to it, then check it off when done.**  Feedback appreciated.

**Comment by LeeDongGeon1996** [comment]
Let's go to the next step, `Add remaining indicators and unit tests`.🚀

**Comment by DaveSkender** [comment]
@LeeDongGeon1996 I've changed my mind about having this in a separate repo.  We're starting a separate repo for a Java version, so it makes sense to do the same for the Python wrapper.  I'll create it and add you and we can work towards moving it over.  Once I get some time, I may do some initial setup or even do the move itself of existing Python code, but may not get to it immediately -- feel free to help.  I'll add you as a full contributor.

**Reply by DaveSkender** [reply]
I've created it and kept it private for now, until we get the basic setup in place and are ready to take on more community contributors.

**Reply by LeeDongGeon1996** [reply]
I like this decision and accepted the invitation to new repo for Python wrapper. 
And I'm planning to get back to this work at least in this month. Need to get used to my new day-routine. (Too many work here🤯)

**Comment by cryptocoinserver** [comment]
How is this going? Do you still need help? As TA-Lib and Tulipy both are more or less dead, and it's a matter of time until they don't work anymore, this would be the perfect replacement library. We use their Python library in a python open-source bot framework. We might have capable python programmers in the community, and I myself might be able to help out here and there a little too.

**Reply by DaveSkender** [reply]
Yes, we’re happy to invite more contributors to the Python wrapper project.  It’s currently in Private visibility, just to get a baseline setup, but I expect we’ll make it public in the next week or so to start including more people.

@LeeDongGeon1996 do you have any opinions on when to make the repo public?  I think we’re close.

**Reply by LeeDongGeon1996** [reply]
Glad to meet new contributors. Yes, I think it would be great to make the repo public after resolving package naming issue. Next week would be the perfect.

**Reply by cryptocoinserver** [reply]
Any updates?

**Reply by DaveSkender** [reply]
I've just made the repo public so more contributors can join.  There's still a bit of a [task list of To Do items](https://github.com/DaveSkender/Stock.Indicators.Python/issues/62) before we are ready to release our initial packages, but any support, feedback, or direct help to get there is much appreciated.  We also have [a Backlog](https://github.com/DaveSkender/Stock.Indicators.Python/projects/1) to keep current and longer-term things prioritized.

**Comment by DaveSkender** [comment]
We’re getting close to parity in the Python wrapper and are mostly working on stabilizing the API and working on user feedback before going to full v1.0.0, in a few months, probably.  @LeeDongGeon1996 has been working overtime catching up and has been doing a great job as maintainer.

Tell your Python friends!
https://python.stockindicators.dev/
